Python/import.c

```diff
@@ -3715,16 +3724,7 @@ _imp_create_dynamic_impl(PyObject *module, PyObject *spec, PyObject *file)
 
     PyThreadState *tstate = _PyThreadState_GET();
     mod = import_find_extension(tstate, name, path);
-    if (mod != NULL) {
-        const char *name_buf = PyUnicode_AsUTF8(name);
-        assert(name_buf != NULL);
-        if (_PyImport_CheckSubinterpIncompatibleExtensionAllowed(name_buf) < 0) {
-            Py_DECREF(mod);
-            mod = NULL;
-        }
-        goto finally;
-    }
-    else if (PyErr_Occurred()) {
+    if (mod != NULL || _PyErr_Occurred(tstate)) {
```

Comment:
Unless I'm reading this wrong, I think you need a `Py_XDECREF(mod)` before jumping to `finally`, to handle the case where `mod != NULL` _and_ and error occurred.

Reply:
Thanks for your attention to detail.  In this case `import_find_extension()` only returns non-NULL if there is no error, so there's no need to `Py_XDECREF()`.

Reply:
Aha. In that case, it sounds like this should be sufficient, no?

```suggestion
    if (mod == NULL) {
```

Reply:
Ah, no forget it, I misread your reply :) Sorry for the noise!

Reply:
The rest of the function handles the `mod == NULL && !_PyErr_Occurred(tstate)` case, so `mod == NULL` wouldn't work.